gobblin-example/src/main/java/org/apache/gobblin/example/githubjsontoparquet/EmbeddedGithubJsonToParquet.java

```diff
@@ -0,0 +1,168 @@
+
+  @Slf4j
+  @Alias(value = "githubjsontoparquet", description = "Extract Github data and write to parquet files")
+  public static class CliFactory extends PublicMethodsGobblinCliFactory {
+
+    public CliFactory() {
+      super(EmbeddedGithubJsonToParquet.class);
+    }
+
+    @Override
+    public EmbeddedGobblin constructEmbeddedGobblin(CommandLine cli)
+        throws JobTemplate.TemplateException, IOException {
+      String[] args = cli.getArgs();
+      if (args.length < 1) {
+        throw new RuntimeException("Unexpected number of arguments.");
+      }
+      try {
+        if (args.length == 2) {
+          return new EmbeddedGithubJsonToParquet(args[0], args[1]);
+        }
+      } catch (JobTemplate.TemplateException | IOException e) {
+        e.printStackTrace();
+      }
+      return null;
+    }
+
+    @Override
+    public String getUsageString() {
+      return "1st arg: Archive to pull ex: 2015-01-01-15 (yyyy-mm-dd-hh); 2nd arg: Work dir with filesystem URI";
+    }
+  }
+
+  @CliObjectSupport(argumentNames = {"archiveDateAndHour", "workDir"})
+  public EmbeddedGithubJsonToParquet(String archiveDateAndHour, String workDir)
+      throws JobTemplate.TemplateException, IOException {
+    super("githubjsontoparquet");
+    URL workDirUrl;
+    try {
+      workDirUrl = new URL(workDir);
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+      throw new RuntimeException("Work directory URI with no protocol or malformed.");
+    }
+
+    // Set configuration
+    String fsProtocol = workDirUrl.getProtocol() + ":///";
+    this.setConfiguration(WORK_DIR_KEY, workDir);
+    this.setConfiguration(ConfigurationKeys.FS_URI_KEY, fsProtocol);
+    this.setConfiguration(ConfigurationKeys.STATE_STORE_ENABLED, "true");
+    this.setConfiguration(ConfigurationKeys.STATE_STORE_ROOT_DIR_KEY, workDir + "/store");
+    this.setConfiguration(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, fsProtocol);
+    this.setConfiguration(ConfigurationKeys.DATA_PUBLISHER_FILE_SYSTEM_URI, fsProtocol);
+    this.setConfiguration(ConfigurationKeys.DATA_PUBLISHER_FINAL_DIR, workDir + "/event_data");
+    this.setConfiguration(ConfigurationKeys.DATA_PUBLISHER_METADATA_OUTPUT_DIR, workDir + "/metadata");
+    this.setConfiguration(ConfigurationKeys.WRITER_OUTPUT_FORMAT_KEY, WriterOutputFormat.PARQUET.toString());
+
+    //Set template
+    try {
+      setTemplate(ResourceBasedJobTemplate.forResourcePath("githubjsontoparquet.template"));
+    } catch (URISyntaxException | SpecNotFoundException e) {
+      e.printStackTrace();
+      throw new RuntimeException("Cannot set template");
+    }
+
+    // Download the archive
+    String fileUrl = String.format(GITHUB_ARCHIVE_URL_TEMPLATE, archiveDateAndHour);
+    Path downloadDirPath = createDownloadDir(workDirUrl.getPath(), fileUrl);
+    Path downloadFile = getAbsoluteDownloadFilePath(downloadDirPath, archiveDateAndHour);
+    if (downloadFile.toFile().exists()) {
```

Comment:
Move line 129 to 136 to `downloadFile` method? They look like some validation logic belongs to `downloadFile`